transportable-udfs-examples/transportable-udfs-example-udfs/src/main/java/com/linkedin/transport/examples/StructElementIncrementByOneFunction.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.examples;
+
+import com.google.common.collect.ImmutableList;
+import com.linkedin.transport.api.data.StdInteger;
+import com.linkedin.transport.api.data.StdStruct;
+import com.linkedin.transport.api.udf.StdUDF1;
+import com.linkedin.transport.api.udf.TopLevelStdUDF;
+import java.util.List;
+
+
+public class StructElementIncrementByOneFunction extends StdUDF1<StdStruct, StdStruct> implements TopLevelStdUDF {
+
+  @Override
+  public List<String> getInputParameterSignatures() {
+    return ImmutableList.of(
+        "row(integer, integer)"
+    );
+  }
+
+  @Override
+  public String getOutputParameterSignature() {
+    return "row(integer, integer)";
+  }
+
+  @Override
+  public StdStruct eval(StdStruct myStruct) {
+    int currVal = ((StdInteger) myStruct.getField(0)).get();
+    myStruct.setField(0, getStdFactory().createInteger(currVal + 1));
```

Comment:
can you try https://github.com/linkedin/transport/blob/09a89508296a2491f43cc8866d47952c911313ab/transportable-udfs-hive/src/main/java/com/linkedin/transport/hive/data/HiveStruct.java#L64 (using field_name instead of field index) and see if the same issue persists?

Reply:
I don't think transport UDFs test framework lets users create structs with named fields. You can try that and let me know if I am mistaken.

Reply:
I tried fixing this in a totally different PR here: https://github.com/linkedin/transport/pull/119

But this issue is independent.